onnxruntime/core/providers/coreml/model/model.mm

```diff
@@ -395,17 +438,38 @@ Status Predict(const std::unordered_map<std::string, OnnxTensorData>& inputs,
       compiled_model_path_ = [compileUrl path];
 
       MLModelConfiguration* config = [[MLModelConfiguration alloc] init];
-
-      if (coreml_compute_unit_ & COREML_FLAG_USE_CPU_ONLY) {
+      uint32_t coreml_compute_unit = coreml_options_.ComputeUnits();
+      if (coreml_compute_unit & COREML_FLAG_USE_CPU_ONLY) {
         config.computeUnits = MLComputeUnitsCPUOnly;
-      } else if (coreml_compute_unit_ & COREML_FLAG_USE_CPU_AND_GPU) {
+      } else if (coreml_compute_unit & COREML_FLAG_USE_CPU_AND_GPU) {
         config.computeUnits = MLComputeUnitsCPUAndGPU;
-      } else if (coreml_compute_unit_ & COREML_FLAG_ONLY_ENABLE_DEVICE_WITH_ANE) {
+      } else if (coreml_compute_unit & COREML_FLAG_ONLY_ENABLE_DEVICE_WITH_ANE) {
         config.computeUnits = MLComputeUnitsCPUAndNeuralEngine;  // Apple Neural Engine
       } else {
         config.computeUnits = MLComputeUnitsAll;
       }
 
+      if (coreml_options_.AllowLowPrecisionAccumulationOnGPU()) {
+        config.allowLowPrecisionAccumulationOnGPU = YES;
+      }
+
+// Set the specialization strategy to FastPrediction  for macOS 10.15+
+#if defined(__APPLE__) && defined(__clang__) && __clang_major__ >= 15
+      if (HAS_COREML8_OR_LATER) {
```

Comment:
why do we need to check the clang version?

is this just for macOS or does it set it for iOS as well?

Reply:
Do you have suggestions on passing the clang-tidy  https://github.com/microsoft/onnxruntime/actions/runs/12118598969/job/33783428976?pr=22975?

Reply:
MLComputePlan and `MLOptimizationHints` is defined since __clang_major__ >= 15

Reply:
Not quite following how a feature is tied to a compiler version. The doco talks about the system versions required not the clang version, so can we use something similar to what you have in ProfileComputePlan and specify the macOS/iOS/macCatalyst versions required?

https://developer.apple.com/documentation/coreml/mloptimizationhints-swift.struct
https://developer.apple.com/documentation/coreml/mlcomputeplan-1w21n

Or maybe define a value based on those for readability?

e.g.

`#define HAS_MLCOMPUTEPLAN @available(macos(14.4), ios(17.4), macCatalyst(17.4))`

Reply:
HAS_COREML8_OR_LATER is somewhat `available` function.
but it's not a macro. However `specializationStrategy` is introduced since clang_major > 15 in `coreml.h`.
I think if there is macro to determinate macos version, that would also work.